### DISCUSSION TITLE
Enhancing Plugin Reference Documentation

### DISCUSSION THREAD

**Discussion by trueberryless** [withastro/starlight#2495] [Feature Requests]
The Plugin Reference Documentation gives a perfect overview of how to integrate with Starlight/ Astro by using hooks. However, especially the section [`addIntegration`](https://starlight.astro.build/reference/plugins/#addintegration) could be enhanced by adding further documentation for the following key points:

1. Install the integration on the consumer by defining the package as a peer dependency in your `package.json` of the plugin.
2. Furthermore it would be great to provide a link to [`peerDependencyMeta`](https://pnpm.io/package_json#peerdependenciesmeta), if the peer dependency should be optional.
3. The code provided could be further improved to import react only after checking if it's not already installed on the consumer. 

I want to mention and thank @ematipico and link to the [original issue in the Astro repo](https://github.com/withastro/astro/issues/12253).

**Comment by HiDeoo** [comment]
Pasting a few thoughts I shared on Discord to not forget them:

> My first thought is that there is definitely room for improvements altho it could be tricky to get it right as it depends a lot on what the plugin is doing, we don't know if the plugin requires the integration to be a dep or a peer dep, if it's optional or not, etc. I assume we also don't want (and probably cannot) document `peerDependenciesMeta` (maybe it could be a link if we go that far).
